src/Microsoft.AspNetCore.SignalR.Core/JsonProtocolDependencyInjectionExtensions.cs

```diff
@@ -16,7 +17,8 @@ public static class JsonProtocolDependencyInjectionExtensions
         /// </remarks>
         /// <param name="builder">The <see cref="ISignalRBuilder"/> representing the SignalR server to add JSON protocol support to.</param>
         /// <returns>The value of <paramref name="builder"/></returns>
-        public static ISignalRBuilder AddJsonProtocol(this ISignalRBuilder builder) => AddJsonProtocol(builder, _ => { });
+        public static TBuilder AddJsonProtocol<TBuilder>(this TBuilder builder) where TBuilder : ISignalRBuilder
```

Comment:
Why is it generic? Why not just ISignalRBuilder

Reply:
Because the only extension methods that hang off ISignalRBuilder are AddJsonProtocol and AddMessagePackProcotol, and if ISignalRBuilder is returned then you can't chain anymore methods off it, e.g. WithUrl and Build below won't work.

```
var builder = new HubConnectionBuilder().AddHubProtocol().WithUrl(url).Build();
```

Need to return the passed in type using generics.

Reply:
I see, awful 😄